docs/source/format/CanonicalExtensions.rst

```diff
@@ -72,4 +72,30 @@ same rules as laid out above, and provide backwards compatibility guarantees.
 Official List
 =============
 
-No canonical extension types have been standardized yet.
+Fixed shape tensor
+==================
+
+* Extension name: `arrow.fixed_shape_tensor`.
+
+* The storage type of the extension: ``FixedSizeList`` where:
+
+  * **value_type** is the data type of individual tensors and
+    is an instance of ``pyarrow.DataType`` or ``pyarrow.Field``.
+  * **list_size** is the product of all the elements in tensor shape.
+
+* Extension type parameters:
+
+  * **value_type** = Arrow DataType of the tensor elements
+  * **shape** = shape of the contained tensors as a tuple
+
+* Description of the serialization:
+
+  The metadata must be a valid JSON object including shape of
+  the contained tensors as an array with key "shape".
+
+  For example: `{ "shape": [2, 5]}`
+
+.. note::
+
+  Elements in an fixed shape tensor extension array are stored
+  in row-major/C-contiguous order.
```

Comment:
Do you have the same problem right now with FixedSizeList arrays? Or what is this converted to in R now? (can you convert that to a single matrix?)

Also, even if we allow a different order or custom strides for each _individual tensor_, the full array backing the FixedSizeListArray (the flat values child array) still needs the first dimension (with size == length of the logical array) with the biggest strides. So if R doesn't support that, I don't think zero-copy conversion is ever possible?

Reply:
On further thinking, I am also not sure this is actually a problem. Languages like Julia and R can still create an array/matrix from the raw memory of the ListArray zero-copy. This array will only look transposed to their users if you need to prevent the copy. It's then up to the application/user to decide if you want to keep that or not (have a strided view in expected dimension order, or create a copy)?

Reply:
Well, if you have only one tensor, then both ways would work anyway, right?